bokehjs/src/lib/core/visuals.ts

```diff
@@ -316,16 +304,14 @@ export class Fill extends ContextProperties {
 
   protected _set_vectorize(ctx: Context2d, i: number): void {
     this.cache_select("fill_color", i)
-    if (ctx.fillStyle !== this.cache.fill_color)
-      ctx.fillStyle = this.cache.fill_color
+    ctx.fillStyle = this.cache.fill_color
```

Comment:
eventually, something this would probably be ideal to work towards:
```
ctx.fillStyle = this.cache.fill_color(i)
```
to remove all the clunky explicit `cache_select` calls

Maybe there is an even better way to handle the scalar/vector split even more efficiently.

Reply:
I started gearing towards a cleaner separation between scalars and vectors in the last commit of PR #10004. I will address this in near future.